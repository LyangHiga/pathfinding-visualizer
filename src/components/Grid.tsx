import React from "react";

import Node from "./Node";

import { GridProps } from "./types/GridTypes";

function Grid(props: GridProps) {
  const {
    grid,
    isWeighted,
    handleMouseDown,
    handleMouseEnter,
    handleMouseUp,
  } = props;

  return (
    <div style={{ textAlign: "center" }}>
      {grid.grid.map((row, rowIdx) => {
        return (
          <div key={rowIdx}>
            {row.map((node) => {
              const { row, col, val, weight } = node;
              return (
                <Node
                  key={val}
                  val={val}
                  col={col}
                  row={row}
                  onMouseDown={handleMouseDown}
                  onMouseEnter={handleMouseEnter}
                  onMouseUp={handleMouseUp}
                  weight={weight}
                  isWeighted={isWeighted}
                ></Node>
              );
            })}
          </div>
        );
      })}
    </div>
  );
}

export default Grid;
