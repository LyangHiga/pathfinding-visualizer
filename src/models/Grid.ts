import Node from "./Node";
import { createAdjList, getVal } from "../helpers/gridHelper";

export default class Grid {
  public grid: Node[][];
  public start: number;
  public target: number;
  public readonly nRows: number;
  public readonly nCols: number;
  // min and max weight allowed
  public readonly min: number;
  public readonly max: number;
  constructor(
    start: number,
    target: number,
    nRows: number,
    nCols: number,
    // min and max weight
    max = 1,
    min = 1
  ) {
    this.grid = [];
    this.start = start;
    this.target = target;
    this.nCols = nCols;
    this.nRows = nRows;
    this.min = min;
    this.max = max;
    for (let row = 0; row < nRows; row++) {
      const currentRow: Node[] = [];
      for (let col = 0; col < nCols; col++) {
        // get random weight for all edges of this node
        const weight = Math.floor(Math.random() * (max - min) + min);
        const val = getVal(row, nCols, col);
        const AdjList = createAdjList(val, col, row, nRows, nCols);
        const newNode = new Node(col, row, val, start, target, AdjList, weight);
        currentRow.push(newNode);
      }
      this.grid.push(currentRow);
    }
  }
}
