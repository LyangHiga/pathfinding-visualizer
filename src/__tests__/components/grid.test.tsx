import React from "react";
import { shallow } from "enzyme";
import Grid from "../../components/Grid";

import GridModel from "../../models/Grid";

const START = 10;
const TARGET = 14;

const NCOLS = 5;
const NROWS = 5;

//       0  1   2   3   4
//       5  6   7   8   9
// g =   S  11  12  13  T
//      15  16  17  18  19
//      20  21  22  23  24
describe("Grid", () => {
  it("renders correctly", () => {
    const g = new GridModel(START, TARGET, NROWS, NCOLS);
    const wrapper = shallow(
      <Grid
        grid={g}
        isWeighted={true}
        handleMouseDown={jest.fn()}
        handleMouseEnter={jest.fn()}
        handleMouseUp={jest.fn()}
      />
    );
    expect(wrapper).toMatchSnapshot();
    // On the first run of this test, Jest will generate a snapshot file automatically.
  });
});
