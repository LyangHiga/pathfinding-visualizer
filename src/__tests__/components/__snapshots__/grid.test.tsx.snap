// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid renders correctly 1`] = `
<div
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <div
    key="0"
  >
    <Node
      col={0}
      isWeighted={true}
      key="0"
      onMouseDown={[MockFunction]}
      onMouseEnter={[MockFunction]}
      onMouseUp={[MockFunction]}
      row={0}
      val={0}
      weight={1}
    />
    <Node
      col={1}
      isWeighted={true}
      key="1"
      onMouseDown={[MockFunction]}
      onMouseEnter={[MockFunction]}
      onMouseUp={[MockFunction]}
      row={0}
      val={1}
      weight={1}
    />
    <Node
      col={2}
      isWeighted={true}
      key="2"
      onMouseDown={[MockFunction]}
      onMouseEnter={[MockFunction]}
      onMouseUp={[MockFunction]}
      row={0}
      val={2}
      weight={1}
    />
    <Node
      col={3}
      isWeighted={true}
      key="3"
      onMouseDown={[MockFunction]}
      onMouseEnter={[MockFunction]}
      onMouseUp={[MockFunction]}
      row={0}
      val={3}
      weight={1}
    />
    <Node
      col={4}
      isWeighted={true}
      key="4"
      onMouseDown={[MockFunction]}
      onMouseEnter={[MockFunction]}
      onMouseUp={[MockFunction]}
      row={0}
      val={4}
      weight={1}
    />
  </div>
  <div
    key="1"
  >
    <Node
      col={0}
      isWeighted={true}
      key="5"
      onMouseDown={[MockFunction]}
      onMouseEnter={[MockFunction]}
      onMouseUp={[MockFunction]}
      row={1}
      val={5}
      weight={1}
    />
    <Node
      col={1}
      isWeighted={true}
      key="6"
      onMouseDown={[MockFunction]}
      onMouseEnter={[MockFunction]}
      onMouseUp={[MockFunction]}
      row={1}
      val={6}
      weight={1}
    />
    <Node
      col={2}
      isWeighted={true}
      key="7"
      onMouseDown={[MockFunction]}
      onMouseEnter={[MockFunction]}
      onMouseUp={[MockFunction]}
      row={1}
      val={7}
      weight={1}
    />
    <Node
      col={3}
      isWeighted={true}
      key="8"
      onMouseDown={[MockFunction]}
      onMouseEnter={[MockFunction]}
      onMouseUp={[MockFunction]}
      row={1}
      val={8}
      weight={1}
    />
    <Node
      col={4}
      isWeighted={true}
      key="9"
      onMouseDown={[MockFunction]}
      onMouseEnter={[MockFunction]}
      onMouseUp={[MockFunction]}
      row={1}
      val={9}
      weight={1}
    />
  </div>
  <div
    key="2"
  >
    <Node
      col={0}
      isWeighted={true}
      key="10"
      onMouseDown={[MockFunction]}
      onMouseEnter={[MockFunction]}
      onMouseUp={[MockFunction]}
      row={2}
      val={10}
      weight={1}
    />
    <Node
      col={1}
      isWeighted={true}
      key="11"
      onMouseDown={[MockFunction]}
      onMouseEnter={[MockFunction]}
      onMouseUp={[MockFunction]}
      row={2}
      val={11}
      weight={1}
    />
    <Node
      col={2}
      isWeighted={true}
      key="12"
      onMouseDown={[MockFunction]}
      onMouseEnter={[MockFunction]}
      onMouseUp={[MockFunction]}
      row={2}
      val={12}
      weight={1}
    />
    <Node
      col={3}
      isWeighted={true}
      key="13"
      onMouseDown={[MockFunction]}
      onMouseEnter={[MockFunction]}
      onMouseUp={[MockFunction]}
      row={2}
      val={13}
      weight={1}
    />
    <Node
      col={4}
      isWeighted={true}
      key="14"
      onMouseDown={[MockFunction]}
      onMouseEnter={[MockFunction]}
      onMouseUp={[MockFunction]}
      row={2}
      val={14}
      weight={1}
    />
  </div>
  <div
    key="3"
  >
    <Node
      col={0}
      isWeighted={true}
      key="15"
      onMouseDown={[MockFunction]}
      onMouseEnter={[MockFunction]}
      onMouseUp={[MockFunction]}
      row={3}
      val={15}
      weight={1}
    />
    <Node
      col={1}
      isWeighted={true}
      key="16"
      onMouseDown={[MockFunction]}
      onMouseEnter={[MockFunction]}
      onMouseUp={[MockFunction]}
      row={3}
      val={16}
      weight={1}
    />
    <Node
      col={2}
      isWeighted={true}
      key="17"
      onMouseDown={[MockFunction]}
      onMouseEnter={[MockFunction]}
      onMouseUp={[MockFunction]}
      row={3}
      val={17}
      weight={1}
    />
    <Node
      col={3}
      isWeighted={true}
      key="18"
      onMouseDown={[MockFunction]}
      onMouseEnter={[MockFunction]}
      onMouseUp={[MockFunction]}
      row={3}
      val={18}
      weight={1}
    />
    <Node
      col={4}
      isWeighted={true}
      key="19"
      onMouseDown={[MockFunction]}
      onMouseEnter={[MockFunction]}
      onMouseUp={[MockFunction]}
      row={3}
      val={19}
      weight={1}
    />
  </div>
  <div
    key="4"
  >
    <Node
      col={0}
      isWeighted={true}
      key="20"
      onMouseDown={[MockFunction]}
      onMouseEnter={[MockFunction]}
      onMouseUp={[MockFunction]}
      row={4}
      val={20}
      weight={1}
    />
    <Node
      col={1}
      isWeighted={true}
      key="21"
      onMouseDown={[MockFunction]}
      onMouseEnter={[MockFunction]}
      onMouseUp={[MockFunction]}
      row={4}
      val={21}
      weight={1}
    />
    <Node
      col={2}
      isWeighted={true}
      key="22"
      onMouseDown={[MockFunction]}
      onMouseEnter={[MockFunction]}
      onMouseUp={[MockFunction]}
      row={4}
      val={22}
      weight={1}
    />
    <Node
      col={3}
      isWeighted={true}
      key="23"
      onMouseDown={[MockFunction]}
      onMouseEnter={[MockFunction]}
      onMouseUp={[MockFunction]}
      row={4}
      val={23}
      weight={1}
    />
    <Node
      col={4}
      isWeighted={true}
      key="24"
      onMouseDown={[MockFunction]}
      onMouseEnter={[MockFunction]}
      onMouseUp={[MockFunction]}
      row={4}
      val={24}
      weight={1}
    />
  </div>
</div>
`;
